.github/workflows/cross.yml

```diff
@@ -132,45 +132,80 @@ jobs:
+      matrix:
+        target:
+          - 'x86_64-pc-windows-gnu'
+          # TODO: Fails in GitHub CI.  Appears to be related to bindgen.
+          # .../base.h:61:10: fatal error: 'stdlib.h' file not found
+          # - 'i686-pc-windows-gnu'
     steps:
       - uses: ilammy/setup-nasm@v1
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           submodules: 'recursive'
       - uses: dtolnay/rust-toolchain@master
         id: toolchain
         with:
           toolchain: stable
-          target: x86_64-pc-windows-gnu
-      - name: Debug Test on `x86_64-pc-windows-gnu`
-        run: cargo test -p aws-lc-rs --target x86_64-pc-windows-gnu --features bindgen
-      - name: Release test on `x86_64-pc-windows-gnu`
-        run: cargo test -p aws-lc-rs --release --target x86_64-pc-windows-gnu --features bindgen
+          target: ${{ matrix.target }}
+      - name: Debug Test on `${{ matrix.target }}`
+        run: cargo test -p aws-lc-rs --target ${{ matrix.target }} --features bindgen
+      - name: Release test on `${{ matrix.target }}`
+        run: cargo test -p aws-lc-rs --release --target ${{ matrix.target }} --features bindgen
 
-  aws-lc-rs-windows-arm64:
-    if: github.repository_owner == 'aws'
-    name: aarch64-pc-windows-msvc
+  aws-lc-rs-windows-msvc:
+    if: ${{ github.repository_owner == 'aws' }}
+    name: ${{ matrix.target }} - ${{ (matrix.crt_static == '1' && 'crt-static and /WX') || '/WX' }}
     runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - 'x86_64-pc-windows-msvc'
+          - 'i686-pc-windows-msvc'
+          - 'aarch64-pc-windows-msvc'
+        crt_static:
+          - '0'
+          - '1'
     steps:
-      - uses: actions/checkout@v3
+      - name: Install NASM
+        if: ${{ matrix.target == 'x86_64-pc-windows-msvc' || matrix.target == 'i686-pc-windows-msvc' }}
+        uses: ilammy/setup-nasm@v1
+      - uses: actions/checkout@v4
         with:
           submodules: 'recursive'
-      - name: Install ninja-build tool
-        uses: seanmiddleditch/gha-setup-ninja@v4
       - uses: dtolnay/rust-toolchain@master
         id: toolchain
         with:
           toolchain: stable
-          target: aarch64-pc-windows-msvc
-      - name: Build for `aarch64-pc-windows-msvc`
-        run: cargo build -p aws-lc-rs --target aarch64-pc-windows-msvc --features bindgen
+          target: ${{ matrix.target }}
+      - name: Set ACTION_CARGO
+        run: |
+          if ('${{ matrix.target }}' -like '*aarch64*') {
+            echo "ACTION_CARGO=build" | Out-File -FilePath $env:GITHUB_ENV -Append
+          } else {
+            echo "ACTION_CARGO=test" | Out-File -FilePath $env:GITHUB_ENV -Append
+          }
+      - name: Set RUSTFLAGS
+        run: |
+          if ('${{ matrix.crt_static }}' -eq '1') {
+            echo "RUSTFLAGS=-Clink-arg=/WX -Ctarget-feature=+crt-static" | Out-File -FilePath $env:GITHUB_ENV -Append
```

Comment:
I might have missed this, but what does `-Clink-arg=/WX` actually do?

Reply:
The "[/WX](https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170)" force linker warning to surface as errors in the build.  It helps us avoid having "false negatives" where our Windows build succeeds even though there are problems with the linkage.